packages/block-library/src/navigation-link/index.php

```diff
@@ -322,13 +322,32 @@ function build_variation_for_navigation_link( $entity, $kind ) {
 	return $variation;
 }
 
+/**
+ * Register a variation for a post type / taxonomy for the navigation link block
+ *
+ * @param array $variation Variation array from build_variation_for_navigation_link.
+ * @return void
+ */
+function register_block_core_navigation_link_variation( $variation ) {
+	// Directly set the variations on the registered block type
+	// because there's no server side registration for variations (see #47170).
+	$navigation_block_type = WP_Block_Type_Registry::get_instance()->get_registered( 'core/navigation-link' );
+	if ( ! $navigation_block_type ) {
+		return;
+	}
+
+	$navigation_block_type->variations[] = $variation;
```

Comment:
This makes me think we should expose a method on [`WP_Block_Type`](https://developer.wordpress.org/reference/classes/wp_block_type/) for this purpose. 

One for a follow up Issue I think.

Reply:
There's already #47170 about registering variations on the server. Allthough for block styles in `register_block_style` it does not operate on the `WP_Block_Type` directly, but instead uses a `WP_Block_Styles_Registry`.